### PR TITLE
LPS-98886 AB Test variants (DO NOT BACKPORT)

### PR DESCRIPTION
**Pull request by darquesdev**
@dgarciasarai @andresfulla @ealonso @jkappler

### CODE REVIEW

**Comment by darquesdev** [comment]
ci:test:sf

**Comment by darquesdev** [comment]
ci:test:relevant

**Comment by epgarcia** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentLocalServiceImpl.java [hunk 1]
```diff
@@ -105,6 +105,11 @@ public SegmentsExperiment deleteSegmentsExperiment(
 
 		segmentsExperimentPersistence.remove(segmentsExperiment);
 
+		// Segments experiment rels
```
Comment (darquesdev) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentLocalServiceImpl.java]: this could be included in another commit

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/model/impl/SegmentsExperimentRelImpl.java [hunk 2]
```diff
@@ -28,12 +35,25 @@
...
-	/**
-	 * NOTE FOR DEVELOPERS:
-	 *
-	 * Never reference this class directly. All methods that expect a segments experiment rel model instance should use the {@link com.liferay.segments.model.SegmentsExperimentRel} interface instead.
-	 */
 	public SegmentsExperimentRelImpl() {
 	}
 
+	@Override
+	public String getSegmentsExperienceName(Locale locale)
```
Comment (darquesdev) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/model/impl/SegmentsExperimentRelImpl.java]: It's been created in order to avoid duplicating code

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/AddSegmentsExperienceMVCActionCommand.java [hunk 3]
```diff
@@ -137,42 +143,124 @@ private JSONObject _addSegmentsExperience(ActionRequest actionRequest)
...
+
+		SegmentsExperience segmentsExperience = _addSegmentsExperience(
+			actionRequest, classNameId, classPK, segmentsExperiment);
 
 		_populateSegmentsExperienceJSONObject(jsonObject, segmentsExperience);
 
 		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
 			WebKeys.THEME_DISPLAY);
 
+		long baseSegmentsExperienceId = _getBaseSegmentsExperienceId(
```
Comment (darquesdev) [inline modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/AddSegmentsExperienceMVCActionCommand.java]: please, review this name

File: modules/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/util/SegmentsExperimentUtil.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.segments.model.SegmentsExperiment;
+import com.liferay.segments.model.SegmentsExperimentRel;
+
+import java.util.Locale;
+
+/**
+ * @author David Arques
+ */
+public class SegmentsExperimentUtil {
```
Comment (darquesdev) [inline modules/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/util/SegmentsExperimentUtil.java]: As in [CommentUtil](https://github.com/liferay/liferay-portal/blob/master/modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/comment/CommentUtil.java)

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentLocalServiceImpl.java [hunk 5]
```diff
@@ -87,6 +87,12 @@ public SegmentsExperiment addSegmentsExperiment(
 
 		segmentsExperimentPersistence.update(segmentsExperiment);
 
+		// Segments experiment rel
```
Comment (darquesdev) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentLocalServiceImpl.java]: This commit could have been be included previously

**Comment by epgarcia** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/76721. [See changes here.](https://github.com/epgarcia/liferay-portal/compare/b79b9eecc003~2...b79b9eecc003) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*
